src/awkward/_v2/contents/listarray.py

```diff
@@ -264,6 +269,13 @@ def _getitem_next_jagged(self, slicestarts, slicestops, slicecontent, tail):
             outoffsets = ak._v2.index.Index64.empty(
                 slicestarts.length + 1, self._nplike
             )
+            # assert (
+            #     outoffsets.nplike is self._nplike
+            #     and slicestarts.nplike is self._nplike
+            #     and slicestops.nplike is self._nplike
+            #     and self._starts.nplike is self._nplike
+            #     and self._stops.nplike is self._nplike
+            # )
```

Comment:
Some of these are commented out (here and following). What's happening?

Is it because the `slicer` in

```python
array[slicer]
```

might not have the same nplike as the `array`? If that's a possibility, then the nplikes that differ should be converted (`_to_backend`), rather than let the difference be ignored. They're about to be used in the kernel that follows, and their `.data` would point to the wrong place (CPU vs GPU).